tests/test_v2.py

```diff
@@ -194,12 +194,8 @@ def test_create_array_defaults(store: Store):
         )
 
 
-@pytest.mark.parametrize("array_order", ["C", "F"])
-@pytest.mark.parametrize("data_order", ["C", "F"])
-@pytest.mark.parametrize("memory_order", ["C", "F"])
-def test_v2_non_contiguous(
-    array_order: Literal["C", "F"], data_order: Literal["C", "F"], memory_order: Literal["C", "F"]
-) -> None:
```

Comment:
@normanrz you originally wrote this test in https://github.com/zarr-developers/zarr-python/pull/2679, but I don't understand why there are three different types of order, and there's no documentation, or even comments on this test explaining.

I've therefore chosen to be a little bit bold, and for at least v2 data this PR now assumes there is only one order. If we need to deal with different types of orders, I'm certainly open to that, but that increased complexity probably needs to come with some very good user docs, given in zarr-python v2 there was only one type of memory order.

Since you originally wrote this test, I wanted to see what you thought about this PR?

Reply:
array_order = order in the metadata
data_order = order of the numpy array that is used to write
memory_order = order in the runtime config which is used for reading data, ie the order of the output numpy array

At least the first two are necessary for this test. If you want to change the behavior to always read in array_order, then you can drop the memory_order.

Reply:
Maybe I'm missing something, but don't we just need to parametrize with one order? We pass that one order parameter to the array creation routine, and then check it's correct in both the metadata and the order the data is stored in memory, right (which is how the test works currently in this PR)?

Reply:
Consider this: You create a Zarr array with F order. Next you create a numpy array with C order. You write the numpy array in the Zarr array. This test makes sure that the data on-disk is in F order. That's why you need at least 2 orders.

Reply:
Aha, that makes sense - thanks!